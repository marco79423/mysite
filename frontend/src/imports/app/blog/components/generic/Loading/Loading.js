import React from 'react'
import styled from 'styled-components'
import twisterStyle from '../animation/twisterStyle'
import loadingImage from './img/loading.jpg'

const Base = styled.div`
  margin-top: 10%;
  display: flex;
  flex-direction: column;
  align-items: center;
`

const LoadingImage = styled.div`
  display: inline-block;

  width: 90px;
  height: 90px;
  background: url(${loadingImage}) no-repeat center;
  background-size: 90px 90px;

  border: 0.1em solid #FFFFFF;
  box-shadow: 0 1px 4px rgba(0, 0, 0, 0.15);
  
  ${twisterStyle}
`

const Message = styled.div`
  margin-top: 2rem;
  font-size: 2rem;
  text-align: center;
`

export default class Loading extends React.PureComponent {
  render () {
    return (
      <Base>
        <LoadingImage/>
        <Message>正在努力讀取中…</Message>
      </Base>
    )
  }
}
