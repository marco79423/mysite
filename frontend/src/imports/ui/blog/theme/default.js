const colorTable = {
  'cyan-50': '#E0F7FA',
  'cyan-600': '#00ACC1',
  'cyan-700': '#0097A7',
  'cyan-800': '#00838F',
  'cyan-900': '#006064',
  'cyan-A700': '#00B8D4',

  'teal-50':'#E0F2F1',
  'teal-100':'#B2DFDB',
  'teal-200':'#80CBC4',
  'teal-300':'#4DB6AC',
  'teal-400':'#26A69A',
  'teal-500':'#009688',
  'teal-600':'#00897B',
  'teal-700':'#00796B',
  'teal-800':'#00695C',
  'teal-900':'#004D40',
  'teal-A700':'#00BFA5',

  'grey-50': '#FAFAFA',
  'grey-200': '#EEEEEE',
  'grey-300': '#E0E0E0',
  'grey-600': '#757575',
  'grey-700': '#616161',
  'grey-900': '#212121'
}

export default {
  global: {
    fontColor: colorTable['grey-900'],
    link: {
      color: colorTable['cyan-700'],
      hoverColor: colorTable['cyan-900']
    }
  },
  page: {
    background: colorTable['grey-700'],
    header: {
      background: colorTable['teal-400'],
      titleColor: colorTable['grey-50']
    },
    nav: {
      background: colorTable['teal-600'],
      menu: {
        color: '#ECECEC',
        background: colorTable['teal-600'],
        hoverBackground: colorTable['cyan-900'],
        dropdown: {
          background: colorTable['teal-600'],
          borderColor: colorTable['teal-800']
        }
      }
    },
    main: {
      background: colorTable['grey-200'],
      content: {
        article: {
          background: 'white',
          metadata: {
            tip: {
              color: '#FFFFFF',
              background: colorTable['teal-600']
            }
          },
          header: {
            color: colorTable['grey-700']
          },
          block: {
            color: 'white',
            background: colorTable['teal-600'],
            borderColor: '#ccc'
          },
          doctestBlock: {
            color: 'white',
            background: '#6D6D6D',
          },
          note: {
            background: '#ECECEC',
            titleColor: colorTable['teal-800'],
            borderColor: '#ccc'
          },
          table: {
            header: {
              background: colorTable['teal-600'],
            }
          }
        },
        pagination: {
          color: colorTable['grey-200'],
          background: colorTable['teal-700'],
          hoverBackground: colorTable['teal-900'],
          activeBackground: colorTable['teal-900'],
        }
      },
      aside: {
        background: colorTable['grey-200'],
        darkerBackground: colorTable['grey-600'],

        section: {
          titleColor: colorTable['grey-700'],
          borderColor: colorTable['grey-200']
        }
      }
    },
    footer: {
      background: colorTable['teal-600'],
      copyrightColor: '#ECECEC',
    }
  }
}