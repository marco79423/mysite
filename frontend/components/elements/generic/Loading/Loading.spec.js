import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {ThemeProvider} from 'styled-components'

import theme from '../../../theme/default'
import Loading from './Loading'

test('It should render Loading component correctly', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Loading/>
    </ThemeProvider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
