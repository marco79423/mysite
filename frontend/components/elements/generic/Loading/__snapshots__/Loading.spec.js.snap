// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Loading component correctly 1`] = `
.c0 {
  background: white;
  padding: 32px;
  min-height: 400px;
}

.c1 {
  margin-top: 10%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: inline-block;
  width: 90px;
  height: 90px;
  background: url(/img/loading.jpg) no-repeat center;
  background-size: 90px 90px;
  border: 0.1em solid #FFFFFF;
  box-shadow: 0 1px 4px rgba(0,0,0,0.15);
  -webkit-animation-name: bQMHby;
  animation-name: bQMHby;
  -webkit-animation-duration: 5s;
  animation-duration: 5s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  z-index: 1000;
}

.c3 {
  margin-top: 2rem;
  font-size: 2rem;
  text-align: center;
}

<article
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c3"
    >
      正在努力讀取中…
    </div>
  </div>
</article>
`;
